README.md

```diff
@@ -1,3 +1,189 @@
+Mattermost Genesis is a tool meant to smooth Mattermost Cloud enterprise adoption. It provides a service to create isolated AWS accounts, provision the networking inrastructure and prepare the ground for Mattermost cloud cluster creation. It offers CIDR pool storage functionality and future releases will include features like VPC peering automation.
+
+## Overview
+
+The following features are supported currently by Genesis:
+- Creation a CIDR pool from provided parent subnets.
+- Creation of AWS accounts.
+- Provisioning of AWS accounts with all necessary infrastructure:
+    - VPCs, Subnets, Route tables
+    - TGWs, NATs, DHCP options
+    - Security Groups
+    - DB parameter groups
+    - IAM resources
+- Provisioning with preselected CIDR or random picked one.
+- Deletion of infrastructure and AWS accounts.
+- Ability to list and get accounts, subnets, parentsubnets.
+
+The following features will be added to Genesis later:
+- Genesis VPC peering feature.
+- Automatic registration of new accounts with Mattermost OU groups to enable SSO fast login.
+- Ability to clean account from resources without deleting it.
+- Integration with Mattermost cloud provisioner.
+
+## Developing
+
+### Environment Setup
+
+#### Required Software
+
+The following is required to properly run the Genesis server.
+
+##### Note: when versions are specified, it is extremely important to follow the requirement. Newer versions will often not work as expected
+
+1. Install [Go](https://golang.org/doc/install)
+2. Install [Terraform](https://learn.hashicorp.com/terraform/getting-started/install.html) version v0.13.5
+3. Install [golang/mock](https://github.com/golang/mock#installation) version 1.4.x
+
+#### Other Setup
+
+1. Specify the region in your AWS config, e.g. `~/.aws/config`:
+```
+[profile genesis]
+region = us-east-1
+```
+2. Generate an AWS Access and Secret key pair, then export them in your bash profile:
+  ```
+  export AWS_ACCESS_KEY_ID=YOURACCESSKEYID
+  export AWS_SECRET_ACCESS_KEY=YOURSECRETACCESSKEY
+  export AWS_PROFILE=genesis
+  ```
+
+3. Create an S3 bucket to store the terraform state.
+  ```bash
+  aws s3api create-bucket --bucket terraform-genesis-state-bucket-<env> --region us-east-1
+  ```
+
+4. Clone this repository into your GOPATH (or anywhere if you have Go Modules enabled)
+
+### Building
+
+Simply run the following:
+
+```bash
+go install ./cmd/genesis
+alias genesis='$HOME/go/bin/genesis'
+```
+
+
+### Running
+Before running the server the first time you must set up the DB with:
+
+```bash
+$ genesis schema migrate
+```
+
+Run the server with all required flags. See list below:
+
+```
+genesis server
+--control-tower-account <the account id of the account tha manages control tower>
+--control-tower-role <the iam role that will be used in the control tower account>
+--managed-ou <the name of the OU used for new accounts>
```

Comment:
nit: Maybe we could make it clearer what an OU is?

Reply:
thanks fixed